zstacklib/zstacklib/utils/gpu.py

```
@@ -996,3 +1015,549 @@ def parse_alibaba_ppu_metric_output(output):
+        if matched_npu_id:
+            result["npuId"] = matched_npu_id
+            try:
+                is_isolated = check_huawei_npu_is_isolated(
+                    matched_npu_id, npu_ids)
+                result["isIsolated"] = is_isolated
+            except Exception as ex:
+                result["isIsolated"] = False
+
+        # Collect product name
+        try:
+            r, o, e = bash_roe(get_huawei_gpu_product_name_cmd(npu_ids))
```

Ruff (0.14.13)

`bash_roe` may be undefined, or defined from star imports

(F405)